### ISSUE TITLE
Japanese cannot be changed to italic text by wiki syntax

### ISSUE THREAD

**Issue by Michaelsy** [dokuwiki/dokuwiki#4080]
In the forum thread ["Japanese cannot be changed to italic text by wiki syntax"](https://forum.dokuwiki.org/d/21627-japanese-cannot-be-changed-to-italic-text-by-wiki-syntax) a user reports about this very behavior.
Further details and aspects can be read in the said thread.

\- Michael Sy.

**Comment by fiwswe** [comment]
I can not reproduce this.

The following works fine and exactly as expected here:
```
斜体

//斜体//

//''斜体''//
```

DW 2023-04-04a, sprintDoc Template, macOS 13.6, Safari 17.0
[dokuwiki.org](https://dokuwiki.org), macOS 13.6, Safari 17.0

I am guessing this is a font rendering issue on the client.

See: https://www.dokuwiki.org/sandbox:issue_4080

**Comment by fiwswe** [comment]
Also works fine on macOS Sonoma 14.0 with Safari 17.0.

**Comment by fiwswe** [comment]
Also works fine on iOS 17.0.3 using the built-in Safari.

**Comment by fiwswe** [comment]
I don't think DokuWiki should start including fonts.
- They would only be needed on some clients.
- This would bloat the DokuWiki installation, not to mention all of the potential licensing issues.
- In the case of WebFonts this would penalise everyone by generating additional network traffic, potentially leaking information to the hoster of the WebFont, etc. As someone who is using OSes where this is a non-issue, I'd be miffed if this happened.
- I do not know of any client that is able to display **all** valid UNICODE code points, let alone in different stylistic variations. Do we want to include fonts that include glyphs for all valid UNICODE code points to fix this? And if so for which version of UNICODE? You know, someone may e.g. want to use Lycian characters in their Wiki and my clients lack glyphs for that out of the box. I'd say no! (I know that Japanese is much more mainstream than Lycian ;-) But this just emphasises that this is a client problem. A modern client that can't handle Japanese text is somewhat lacking.)

That said, templates might include fonts and some do to my knowledge. But in this case I would not recommend that approach. I expect that this issue is not limited to Japanese/Kanji in particular. We are way past the times where the client needed to define codepages to map the limited set of 7 or 8-bit codes to glyphs, thereby limiting the display to a few languages at a time. I'd like to be able to mix languages (and glyphs) on a single page. If my OS/browser combination can't handle that then I need to fix that. So a better approach might be to document ways the user can fix their clients (if that is possible). One solution might be for them to install additional fonts locally on the client. Or a Wiki admin could add WebFonts by modifying `./conf/userstyle.css` if the problem impacts a significant number of that Wiki's users.

In summary: I don't think this is a DokuWiki bug. The syntax works fine and apparently only certain clients seem to have trouble correctly displaying the otherwise valid HTML.

**Comment by obw** [comment]
But I also think, something like that should be documented...

Now after the Problem is found and clearly not inside DokuWiki! SO the users have a quick way to find the solution, when needed, also if it pops up again, RTFM and Link to the solution;)

**Comment by splitbrain** [comment]
The Arial based font stack has been picked deliberately, because back in the day Arial was one of the only fonts covering nearly all of the Unicode supported scripts. IIRC back then the full version of the font clocked in at about 20MB and was automatically installed with Microsoft Office. I think Windows and Internet Explorer only had a smaller version by default, but I am not sure about it. The full font was also part of the web essentials font pack - a free download from Microsoft. This way it was (and is) possible to install the full Arial version on other platforms.

This ticket leaves a couple of questions I'd like to see answered:

* On what systems (barebones as they are installed by default without installing any additional software and fonts) is the problem reproducible?
* I assume the problem would not be reproducible on systems that have been installed with a Japanese locale?
* Am I correctly remembering that there are different sized Arial fonts (one full Unicode version and possibly many localized subsets) and that Windows only installs the smaller versions by default?
* is the Arial based font stack still the best choice for wide Unicode support on most systems (without relying on web fonts)?
* how well supported is that font stack on modern mobile phones?

**Comment by fiwswe** [comment]
> On what systems (barebones as they are installed by default without installing any additional software and fonts) is the problem reproducible?

From what the OP in the Forum wrote I am guessing some version of Microsoft Windows and some unknown versions of Chrome and Edge had problems? But someone would need to provide the specific details. And as all the mentioned browsers are also available on e.g. macOS, my guess could be wrong.

Like I wrote, the issue is not reproducible on Apple OSes. (But that does not mean that something similar could not happen for some other obscure UNICODE glyphs on these OSes. And I have not tested other browsers on macOS/iOS.)

>I assume the problem would not be reproducible on systems that have been installed with a Japanese locale?

I don't know about Microsoft Windows but on macOS/iOS/iPadOS/watchOS the preferred language of the user has no bearing on the supported character set/glyphs. As for Linux and other UNIX OSes, my experience with GUIs is very limited so I have no opinion.

>is the Arial based font stack still the best choice for wide Unicode support on most systems (without relying on web fonts)?

At least the Apple OSes have complex fallback mechanisms in most situations, substituting glyphs from other fonts if the chosen font does not have the requested glyph. While this does not improve the design aesthetically it at least makes the text readable to best degree possible. But at least one active font needs to contain the glyph. Using CSS the web author can specify their ordered preferences, but what they actually get might depend on the browser/OS.

>Am I correctly remembering that there are different sized Arial fonts (one full Unicode version and possibly many localized subsets) and that Windows only installs the smaller versions by default?

On macOS Sonoma there are multiple versions of Arial installed by default:
* Arial (≈2516+ glyphs, no Japanese or Chinese language support. 4 styles)
* Arial Unicode MS (50377 glyphs, Japanese and Chinese supported, only 1 style)
* and others, such as Arial Narrow, Arial Rounded MT Bold, Arial Hebrew, …

On Apple OSes Arial seems to be the font is chosen for `sans-serif` in web pages in Safari. I just did a comparison and it seems to be Arial indeed.

> how well supported is that font stack on modern mobile phones?

iOS works fine with the current DokuWiki setup. I have no experience with or opinion on Android.

@obw Well, there is always [dokuwiki.org](https://dokuwiki.org) where something like this could be documented. Feel free…

However it should be made clear, which situations actually need special consideration. OS/browser combination, language, installed fonts, template, …? Most people will probably never have any issues.
